src/bsp/src/stm32/hal/src/usb.c

```diff
@@ -0,0 +1,41 @@
+#include "hal/usb.h"
+#include "usbd_cdc_if.h"
+
+uint8_t cbuffUsbData[CBUFF_USB_DATA_SIZE];
+
+uint8_t Usb_HasTxFinished(USBD_CDC_HandleTypeDef* hcdc){
+    bool ret = false;
+    while(hcdc->TxState != 0){}
+    if (hcdc->TxState == 0) {
```

Comment:
why would TxState != 0 here? There is a while just before that loops until it does

Reply:
true, artifacts...